### PR TITLE
Refactor the status sub command

### LINKED ISSUE #203: More complete `sourced status` command
- Based on [proposal](https://github.com/src-d/sourced-ce/issues/109#issuecomment-510940323):

> would it make sense bundling it all under a `status` command that gives you all that is currently "active" or initialized/loaded? E.g.:
> 
> A. `sourced status workdirs` -> List all working directories and indicate the active, as we have today with `sourced workdirs`
> B. `sourced status components` -> show the status of all components, as we have today w/ `sourced status`
> C. `sourced status config` -> to show the config used for the currently active workdir (or the given organization name/file path
> D. `sourced status` -> all of the above

### PR DESCRIPTION
**Pull request by carlosms**
Fix #203.

I couldn't find a way to have `sourced status` be a command and have sub-commands, as initially proposed. So I added `sourced status all`.

```
$ sourced status
Please specify one command of: all, components, config or workdirs
Usage:
  sourced [OPTIONS] status [status-OPTIONS] <command>

Show the list of working directories and the current deployment

Help Options:
  -h, --help                                     Show this help message

[status command options]

    Log Options:
          --log-level=[info|debug|warning|error] Logging level (default: info) [$LOG_LEVEL]
          --log-format=[text|json]               log format, defaults to text on a terminal and json otherwise [$LOG_FORMAT]
          --log-fields=                          default fields for the logger, specified in json [$LOG_FIELDS]
          --log-force-format                     ignore if it is running on a terminal or not [$LOG_FORCE_FORMAT]

Available commands:
  all         Show all the available status information
  components  Show the status of the components containers
  config      Show the configuration for the active working directory
  workdirs    List all working directories
```

```
$ sourced status all
List of all working directories:
  /home/cmartin/repos
  /home/cmartin/repos/gitbase
* bblfsh

Configuration used for the active working directory:

COMPOSE_PROJECT_NAME=srcd-YmJsZnNo
GITBASE_VOLUME_TYPE=volume
GITBASE_VOLUME_SOURCE=gitbase_repositories
GITBASE_SIVA=true
GITHUB_ORGANIZATIONS=bblfsh
GITHUB_TOKEN=xxx
NO_FORKS=true
GITBASE_LIMIT_CPU=7.9
GITCOLLECTOR_LIMIT_CPU=3.9
GITBASE_LIMIT_MEM=14869912780


Status of all components:

            Name                          Command                       State                   Ports         
--------------------------------------------------------------------------------------------------------------
srcd-ymjsznno_bblfsh-web_1     /bin/bblfsh-web -addr :808 ...   Up                      0.0.0.0:9999->8080/tcp
srcd-ymjsznno_bblfsh_1         /tini -- bblfshd                 Up                      0.0.0.0:9432->9432/tcp
srcd-ymjsznno_ghsync_1         /bin/sh -c sleep 10s && gh ...   Up                                            
srcd-ymjsznno_gitbase_1        ./init.sh                        Up                      0.0.0.0:3306->3306/tcp
srcd-ymjsznno_gitcollector_1   /bin/dumb-init -- /bin/sh  ...   Up                                            
srcd-ymjsznno_metadatadb_1     docker-entrypoint.sh postgres    Up                      0.0.0.0:5433->5432/tcp
srcd-ymjsznno_postgres_1       docker-entrypoint.sh postgres    Up                      0.0.0.0:5432->5432/tcp
srcd-ymjsznno_redis_1          docker-entrypoint.sh redis ...   Up                      0.0.0.0:6379->6379/tcp
srcd-ymjsznno_sourced-ui_1     /entrypoint.sh                   Up (health: starting)   0.0.0.0:8088->8088/tcp
```

### CODE REVIEW

File: cmd/sourced/cmd/status.go [hunk 1]
```diff
@@ -2,18 +2,165 @@ package cmd
...
+	}
+
+	active, err := activeWorkdir()
+	if isNotExist(err) {
+		// skip printing the config and components when there is no active dir
+		return nil
+	}
+
+	if err != nil {
+		return err
```
Comment (smacker) [inline cmd/sourced/cmd/status.go]: why you return an error here but in other places just `printRed`? Maybe we should make it consistent?

File: cmd/sourced/cmd/status.go [hunk 2]
```diff
@@ -2,18 +2,165 @@ package cmd
...
+type statusAllCmd struct {
+	Command `name:"all" short-description:"Show all the available status information" long-description:"Show all the available status information"`
+}
+
+func (c *statusAllCmd) Execute(args []string) error {
+	fmt.Print("List of all working directories:\n")
+
+	err := printWorkdirsCmd()
+	if err != nil {
+		printRed(err.Error())
```
Comment (smacker) [inline cmd/sourced/cmd/status.go]: is it okay to return 0 exit code in case of an error?

**Review by smacker — APPROVED** [review]
there is one comment the rest is ok.

**Review by smacker — CHANGES_REQUESTED** [review]
oops. need an update to integration tests.

File: cmd/sourced/cmd/status.go (hunk 2 above)
Comment (carlosms) [reply cmd/sourced/cmd/status.go]: I was going to save the error, continue with the other sub commands, and at the end return the original error.
But then the output gets out of order, and we should also consider what happens if 2 sub commands fail, should we append the error messages etc...

So if we are going to return and error in `Execute`, it's much cleaner to just return it here. The possible errors are all unlikely, and if anything fails for a sub command quite probably the other sub commands will also fail.
Done in 3dba9dc

**Comment by carlosms** [comment]
Tests updated in e0fcf78

File: cmd/sourced/cmd/status.go (hunk 2 above)
Comment (smacker) [reply cmd/sourced/cmd/status.go]: awesome. Thanks!

**Review by smacker — APPROVED** [review]

**Review by se7entyse7en — APPROVED** [review]

**Pull request merged**